### ISSUE TITLE
How to integrate this code to r9y9's wavenet_vocoder ?

### ISSUE THREAD

**Issue by rishikksh20** [syang1993/gst-tacotron#14]
Is there any way to integrate this code with wavenet vocoder ?

**Comment by syang1993** [comment]
Hi, the main part of this model is similar to Tacotron1. We can also add the style embedding part to Tacotron2, then integrate it to wavenet to get better results.

**Comment by rishikksh20** [comment]
ok, I modified r9y9/Tacotron-2 code and add your style embedding code in that and sees hows it's working.

**Comment by rishikksh20** [comment]
@syang1993 what about loss function ?

**Comment by syang1993** [comment]
@rishikksh20 The style token is trained under an unsupervised way, I guess we don't need extra loss unless you have a specific purpose.

**Comment by rishikksh20** [comment]
@syang1993 thanks! Is it possible to integrate Tacotron 1 with wavenet vocoder, as the GST Tacotron paper has mentioned that they have tested it on Wavenet, so I think it is possible.

**Comment by syang1993** [comment]
@rishikksh20 I tried to integrate Tacotron1 with wavenet, but the performance is worse than Tacotron2. Though the paper tested it on wavenet, I guess it's easier to do it with tacotron2.

**Comment by syang1993** [comment]
@rishikksh20 I'm not sure what the "receptive field width" mean in Tacotron1? I did Tacotron 2 before, I guess it doesn't take so much time to train. But actually if you add style embedding and reference encoder to Tacotron 2, it will take more time. And the decoder part in this repo not perfectly match to the paper, I just try to use the style embedding idea to see how it works. I guess you may don't need to reconstruct the paper's structure all the same, you can modify it with your own purpose.

**Comment by karamarieliu** [comment]
has anyone tried GST w/ Tacotron 2 and WaveNet? I am working on it now but don't have results yet so this could be all for naught..

**Comment by rishikksh20** [comment]
@karamarieliu could you share your work with me, I am also working on this issue.

**Comment by karamarieliu** [comment]
@rishikksh20 I am currently encounter an evaluation error so I'll post it when that is solved. Rn I have T1 with GST and Wavenet if you wanted that. Still testing it but it runs okay. https://github.com/karamarieliu/gst-tacotron-wavenet

**Comment by rishikksh20** [comment]
@karamarieliu means GST-Tacotron 1 with wavenet_vocoder running fine ?  Do you have any voice sample of that ?
